src/book/02-system/01-nubots/05-vision.mdx

```diff
@@ -5,3 +5,94 @@ title: Vision
+
+These parameters are set for each camera as configuration values in the [Camera module](https://github.com/NUbots/NUbots/tree/master/module/input/Camera/data/config), in each robot's respective folder. The values for the left camera on the robot will be stored in `Left.yaml`. The values for the right camera on the robot will be stored in `Right.yaml`.
+
+The parameters are used in the Camera module to find and set up the cameras. The Camera module emits [Image messages](https://github.com/NUbots/NUbots/tree/master/shared/message/input/Image.proto).
+
+The [projection tool](https://github.com/NUbots/NUbots/tree/master/shared/utility/vision/projection.h), based on [panotools' fisheye projection calculations](https://wiki.panotools.org/Fisheye_Projection), maps a portion of the surface of a sphere to a flat image. The type of projection is specified by the above parameter `len.projection`.
+
+# Visual Mesh
+
+## Algorithm
+
+The [Visual Mesh](https://doi.org/10.1007/978-3-030-27544-0_4) underpins the vision system, and is used for sparse detection of balls, points on the field, field lines, goal posts and other robots.
+
+The Visual Mesh is an input transformation that uses knowledge of a cameras orientation and position relative to an observation plane to greatly increase the performance and accuracy of a convolutional neural network. It utilises the geometry of objects to create a mesh structure that ensures that a similar number of samples points are selected regardless of distance. The result is that networks can be much smaller and simpler while still achieving high levels of accuracy. The Visual Mesh is very capable when it comes to detecting distant objects. Distant objects often become too small for a network to detect accurately, but as the Visual Mesh normalises, its size detections are still accurate. The Visual Mesh can detect soccer balls, field lines and goal posts.
+
+The Visual Mesh is created using the following geometric assumptions for each image:
+
+- The camera lens type, height and orientation with respect to the ground are known.
+- The object is spherical with known radius and remains on the ground.
+
+The height, orientation and radius are used to create a set of unit vectors based at the camera position. Any vector with its origin at the camera can be thought of as a ray of light travelling towards the camera. If the vector is within the field of view of the camera it will be collected by the lens. The vector is mapped to a point on the image using the lens projection equation, provided the point is within the bounds of the image sensor of the camera. The Visual Mesh is formed by taking an array of vectors and associating each of them with a point and pixel in the image. The array of vectors is specially constructed using two equations to efficiently sample the space around the camera for the object. This means the sample points on the image will efficiently sample the image for the object.
```

Comment:
```suggestion
The height, orientation and radius are used to create a set of unit vectors based at the camera position. Any vector with its origin at the camera can be thought of as a ray of light travelling towards the camera. If the vector is within the field of view of the camera it will be collected by the lens. Vectors collected by the camera are mapped to points on the image using the lens projection equation. The Visual Mesh is formed by taking an array of vectors and associating each of them with a, possibly interpolated, pixel in the image. The array of vectors is specially constructed using two equations to efficiently sample the space around the camera for the object. This means the sample points on the image will efficiently sample the image for the object.
```